src/components/Editor/Tabs.js

```diff
@@ -482,7 +489,8 @@ export default connect(
       searchTabs: getSearchTabs(state),
       sourceTabs: getSourcesForTabs(state),
       activeSearch: getActiveSearch(state),
-      searchOn: getActiveSearch(state) === "source"
+      searchOn: getActiveSearch(state) === "source",
+      sourceMetaData: { isReactComponent: true }
```

Comment:
Hard-coded for now so I could complete the rest of the task, what should go here because I can't figure it out? 😢

Reply:
@James300 You can do this

```diff
@@ -484,13 +484,15 @@ class SourceTabs extends PureComponent<Props, State> {

 export default connect(
   state => {
+    const selectedSource = getSelectedSource(state);
+    const sourceId = selectedSource ? selectedSource.get("id") : "";
     return {
-      selectedSource: getSelectedSource(state),
+      selectedSource,
       searchTabs: getSearchTabs(state),
       sourceTabs: getSourcesForTabs(state),
       activeSearch: getActiveSearch(state),
       searchOn: getActiveSearch(state) === "source",
-      sourceMetaData: { isReactComponent: true }
+      sourceMetaData: getSourceMetaData(state, sourceId)
     };
   },
   dispatch => bindActionCreators(actions, dispatch)
```
I think there is still the work to set the sourceMetaData in redux

Reply:
Thank Bomsy this did it 👍